syntax = "proto3";

package piscan_pb;

message Entry {
	uint64 freq = 1;
	string systemTag = 2;
	string entryTag = 3;
	string modulation = 4;
	string entryIndex = 5;
	enum Lockout {
		NONE = 0;		// unlocked
		TIMED = 1;	// timed expiration lockout
		SESSION = 2;	// locked until application stops
		PERSISTENT = 3;	// locked until manually unlocked
	}
	Lockout lockoutType = 6;
	int32 lockoutDuration = 7;	// duration of temporary lockout
	int32 delay = 8;			// scan delay in milliseconds
}

message SignalLevel {
	int32 level = 1;
}

message ScannerContext {
	enum State {
		INVAL = 0;
		SCAN = 1;
		HOLD = 2;
		RECEIVE = 3;
	}
	
	State state = 1;
	Entry entryData = 2;
	bool manualHold = 3;
}

message DemodContext {
	int32 squelch = 1;
	float gain = 2;
}

message GeneralMessage {
	enum Type {
		INFO = 0;
		WARNING = 1;
		ERROR = 2;
		CONSOLE = 3;	// for receiving log messages
	}

	Type type = 1;
	string content = 2;
}

message SystemInfo {
	string version = 1;
	int32 build = 2;
	int32 squelchScaleMin = 3;
	int32 squelchScaleMax = 4;
	repeated string supportedModulations = 5;
}
